Simplify our setup for font metric queries from style

Loirooriol opened the pull request:
This is a backport of https://phabricator.services.mozilla.com/D157589,
by Emilio Cobos Álvarez, plus some additions so that Servo compiles,
and some parts from https://phabricator.services.mozilla.com/D144455.

Should have no change in behavior.

<!-- Please describe your changes on the following line: -->


---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `___` with appropriate data: -->
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [ ] These changes fix #___ (GitHub issue number if applicable)

<!-- Either: -->
- [ ] There are tests for these changes OR
- [X] These changes do not require tests because there should be no change in behavior

<!-- Also, please make sure that "Allow edits from maintainers" checkbox is checked, so that we can help you if you get stuck somewhere along the way.-->

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

Loirooriol commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit a00f1d23016473af06450f245b128855dabfa716 with merge ea1cb339b9698bf9562ef0d61f02a9b8f7553e19...

bors-servo commented:
:sunny: Test successful - [checks-github](https://github.com/servo/servo/actions/runs/4931191266/jobs/8814801561)
State: approved= try=True

Loirooriol committed:
Simplify our setup for font metric queries from style

This is a backport of https://phabricator.services.mozilla.com/D157589,
by Emilio Cobos Álvarez, plus some additions so that Servo compiles,
and some parts from https://phabricator.services.mozilla.com/D144455.

Should have no change in behavior.

Loirooriol commented:
@bors-servo r=mrobinson

bors-servo commented:
:pushpin: Commit ab2ba27 has been approved by `mrobinson`

<!-- @bors-servo r=mrobinson ab2ba273e342c6f1a0c1f29cca3f551fd8926e94 -->

bors-servo commented:
:hourglass: Testing commit ab2ba273e342c6f1a0c1f29cca3f551fd8926e94 with merge 9bec0d0f8e92e90b5d8c99ba691a9d9a6bb41bcc...

bors-servo commented:
:sunny: Test successful - [checks-github](https://github.com/servo/servo/actions/runs/4951576384/jobs/8859633771)
Approved by: mrobinson
Pushing 9bec0d0f8e92e90b5d8c99ba691a9d9a6bb41bcc to master...
